### ISSUE TITLE
VoxelMap World Map chunks are black if they contain endertank <> BC stone fluid pipe <> geothermal generator

### ISSUE THREAD

**Issue by soratidus999** [xJon/Tekkit-2#71]
**Basic details**
- **Are you using a legitimate Minecraft account?**: Yes
- **Are you using Java 8, 64-bit?**: Yes
- **Did you add any additional mods**: yes (detail) 
Optifine
- **Crash/latest log link (if applicable)**: link to log / not applicable
https://mclo.gs/cMR0beD

Blocks that cause issue
https://i.imgur.com/DJ77mrJ.png

Map with unmapped chunks
https://i.imgur.com/pQCIJLm.png

Map with careful use of teleport to identify issue
https://i.imgur.com/khR5UW6.png

**Describe the issue**

Voxelmap is unable to add chunks to the World Map with a specific chain of blocks

The minimap is not affected.

EnterTank (filled with Lava if relevant)
StoneFluidPipe
Geothermal Generator

```
[B#757]  [14:45:46] [Voxelmap WorldMap Calculation Thread 9/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:251]: Exception loading chunk: null
[B#757]  [14:45:46] [Voxelmap WorldMap Calculation Thread 9/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: java.lang.ClassCastException
```

```
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:251]: Exception loading chunk: buildcraft.transport.tile.TilePipeHolder cannot be cast to ic2.core.block.base.tile.TileEntityBlock
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: java.lang.ClassCastException: buildcraft.transport.tile.TilePipeHolder cannot be cast to ic2.core.block.base.tile.TileEntityBlock
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at ic2.core.block.base.BlockMultiID.func_176221_a(BlockMultiID.java:138)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185899_b(BlockStateContainer.java:423)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.PersistentMap.getAndStoreData(PersistentMap.java:281)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.CachedRegion.loadChunkData(CachedRegion.java:531)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.CachedRegion.loadModifiedData(CachedRegion.java:341)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.CachedRegion.access$300(CachedRegion.java:56)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable.doRun(CachedRegion.java:230)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at com.mamiyaotaru.voxelmap.persistent.AbstractNotifyingRunnable.run(AbstractNotifyingRunnable.java:22)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[B#757]  [14:51:05] [Voxelmap WorldMap Calculation Thread 1/INFO]: [com.mamiyaotaru.voxelmap.persistent.CachedRegion$RefreshRunnable:doRun:252]: 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by soratidus999** [comment]
I have tried to ignore `buildcrafttransport:*` and `enderstorage:*` (as ic2:* is already ignored) with VoxelMap-Fixes with no change,

**Comment by ACGaming** [comment]
Known and intended in the current state to prevent any hard crashes. VoxelMap is undergoing several rewrite measures with [VoxelMap Fixes](https://github.com/ACGaming/VoxelMapFixes) to bring it in line with modded environments.
Looks like old caches in this case, make sure to wipe your VoxelMap cache folder in advance and share the first occurring issues.

**Comment by ACGaming** [comment]
@soratidus999 Issue should be fixed with the [latest build of VoxelMap Fixes](https://github.com/ACGaming/VoxelMapFixes/releases/download/1.12.2-1.0.6/VoxelMapFixes-1.12.2-1.0.6.jar). Can you test it again with it?

**Comment by xJon** [comment]
@soratidus999 This should be fixed with https://github.com/ACGaming/VoxelMapFixes/releases/tag/1.12.2-1.0.6, if you want you can already check it out :)

**Comment by xJon** [comment]
Should be fixed in `1.0.5` :)

**Comment by soratidus999** [comment]
confirming fixed, worldmap is rendering as expected in tekkit-2 1.0.5 with VoxelMap Fixes 1.22.20-1.0.6

**Comment by xJon** [comment]
Thanks for confirming!
